src/storage/wal_replay.cpp

```diff
@@ -738,7 +739,7 @@ void WriteAheadLogDeserializer::ReplayRowGroupData() {
 	auto &block_manager = db.GetStorageManager().GetBlockManager();
 	PersistentCollectionData data;
 	deserializer.Set<DatabaseInstance &>(db.GetDatabase());
-	CompressionInfo compression_info(block_manager.GetBlockSize());
+	CompressionInfo compression_info(block_manager.GetBlockSize(), block_manager.GetBlockHeaderSize());
```

Comment:
Instead of passing the block size and block header size everywhere - can we just pass in the `BlockManager` as much as possible and resolve the block size/block header size only at the very end (when we need the value)?